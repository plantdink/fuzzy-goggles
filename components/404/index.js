export * from './Component404';
export { default } from './Component404';
