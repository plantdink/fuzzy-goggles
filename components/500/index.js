export { default } from './Component500';
