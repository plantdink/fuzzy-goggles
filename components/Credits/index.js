export * from './Credits';
export { default } from './Credits';
