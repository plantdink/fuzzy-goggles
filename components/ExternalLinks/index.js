// export { default } from './ExternalLinks';
export * from './ExternalLinks';
