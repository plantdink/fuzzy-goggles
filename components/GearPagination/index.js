export * from './GearPagination';
export { default } from './GearPagination';
