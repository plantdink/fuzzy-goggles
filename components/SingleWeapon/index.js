export * from './SingleWeapon';
export { default } from './SingleWeapon';
