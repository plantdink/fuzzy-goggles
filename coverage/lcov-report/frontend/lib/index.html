
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for frontend/lib</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../index.html">All files</a> frontend/lib</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">84.61% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>55/65</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">64% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>16/25</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">71.42% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>20/28</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">84.78% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>39/46</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="abbreviationsDictionary.js"><a href="abbreviationsDictionary.js.html">abbreviationsDictionary.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="14" class="abs high">14/14</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="12" class="abs high">12/12</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	</tr>

<tr>
	<td class="file high" data-value="displayNumbers.js"><a href="displayNumbers.js.html">displayNumbers.js</a></td>
	<td data-value="90.9" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 90%"></div><div class="cover-empty" style="width: 10%"></div></div>
	</td>
	<td data-value="90.9" class="pct high">90.9%</td>
	<td data-value="11" class="abs high">10/11</td>
	<td data-value="44.44" class="pct low">44.44%</td>
	<td data-value="9" class="abs low">4/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="10" class="abs high">10/10</td>
	</tr>

<tr>
	<td class="file high" data-value="displayStrings.js"><a href="displayStrings.js.html">displayStrings.js</a></td>
	<td data-value="84.61" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 84%"></div><div class="cover-empty" style="width: 16%"></div></div>
	</td>
	<td data-value="84.61" class="pct high">84.61%</td>
	<td data-value="13" class="abs high">11/13</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="85.71" class="pct high">85.71%</td>
	<td data-value="7" class="abs high">6/7</td>
	<td data-value="83.33" class="pct high">83.33%</td>
	<td data-value="12" class="abs high">10/12</td>
	</tr>

<tr>
	<td class="file medium" data-value="testUtils.js"><a href="testUtils.js.html">testUtils.js</a></td>
	<td data-value="74.07" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 74%"></div><div class="cover-empty" style="width: 26%"></div></div>
	</td>
	<td data-value="74.07" class="pct medium">74.07%</td>
	<td data-value="27" class="abs medium">20/27</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="56.25" class="pct medium">56.25%</td>
	<td data-value="16" class="abs medium">9/16</td>
	<td data-value="68.75" class="pct medium">68.75%</td>
	<td data-value="16" class="abs medium">11/16</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Tue Apr 05 2022 07:32:49 GMT+1000 (Australian Eastern Standard Time)
            </div>
        <script src="../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../sorter.js"></script>
        <script src="../../block-navigation.js"></script>
    </body>
</html>
    