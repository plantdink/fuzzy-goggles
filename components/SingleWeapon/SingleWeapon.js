import React from 'react';
import { gql } from '@apollo/client';
import styled from 'styled-components';
import {
  displayPercentage,
  hsDisplay,
  millisecondsToSeconds,
  humanReadableNumber,
} from '../../lib/displayNumbers';
import { deAbbreviate } from '../../lib/abbreviationsDictionary';

export const SingleWeaponStyles = styled.div`
  position: relative;
  display: flex;
  flex-flow: column wrap;
  max-width: var(--maxWidth);
  background: white;
  border: 1px solid var(--offWhite);
  box-shadow: var(--bs);
  padding: 2rem;
  margin-bottom: 3rem;
  font-feature-settings: 'tnum';
  font-variant-numeric: tabular-nums;

  .single-weapon__content {
    display: flex;
    max-width: var(--maxWidth);
    justify-content: center;
    align-items: top;
    gap: 2rem;
    background: white;
    img {
      width: 100%;
      object-fit: contain;
    }
  }

  .single-weapon__sub-content {
    display: flex;
    justify-content: space-between;
  }

  .single-weapon__details {
    flex-direction: column;
    margin-bottom: 2rem;
  }

  .single-weapon__title-bar {
    transform: skew(7deg);
    margin-bottom: 1rem;
  }

  .single-weapon__heading {
    margin: 0 1rem;
    margin-top: -3rem;
    background: var(--orange);
    display: inline;
    line-height: 1.3;
    font-size: clamp(2rem, 5vw, 4rem);
    color: var(--text);
    /* text-shadow: 2px 2px 0 rgba(0, 0, 0, 0.1); */
    padding: 0 1rem;
  }

  .single-weapon__section-heading {
    margin: 0 1rem;
    margin-top: -3rem;
    background: var(--orange);
    display: inline;
    line-height: 1.3;
    font-size: clamp(2rem, 4vw, 3.5rem);
    color: var(--text);
    /* text-shadow: 2px 2px 0 rgba(0, 0, 0, 0.1); */
    padding: 0 1rem;
  }

  .single-weapon__subheading {
    margin: 0 1rem;
    /* margin-top: 0rem; */
    background: var(--orange);
    display: inline;
    line-height: 2;
    font-size: clamp(1rem, 3vw, 2rem);
    font-weight: 600;
    color: var(--text);
    /* text-shadow: 2px 2px 0 rgba(0, 0, 0, 0.1); */
    padding: 0 1rem;
  }

  ul {
    margin: 0;
    color: var(--text);
    font-weight: 400;
    font-size: 1.6rem;
    padding: 0;
    list-style-type: none;
  }

  li {
    margin: 0.75rem 0;
    /* padding: 0.5rem 0 0 1.5rem; */
    padding: 0 2.4rem;
  }

  p {
    color: var(--text);
    margin: 0.75rem 0;
    font-weight: 400;
    padding: 0 2.4rem;
    font-size: 1.6rem;
    line-height: 1.3;
  }

  table {
    table-layout: fixed;
    width: 100%;
    border-collapse: collapse;
    color: var(--text);
    line-height: 1.3;
    font-size: 1.6rem;
    font-variant-numeric: tabular-nums;
  }

  th {
    font-weight: 500;
    /* width: 100%; */
  }

  th,
  td {
    padding: 0.5rem 2rem;
    text-align: left;
    vertical-align: baseline;
  }

  /* thead th:nth-child(1) {
    width: 20%;
  }

  thead th:nth-child(2) {
    width: 20%;
  }

  thead th:nth-child(3) {
    width: 20%;
  }

  thead th:nth-child(4) {
    width: 20%;
  } */

  blockquote {
    color: var(--text);
    margin: 0.85rem 0;
    padding: 0 2.4rem;
    font-weight: 300;
    font-size: 1.6rem;
  }

  small {
    color: var(--text);
    font-size: 1.2rem;
    padding: 0 2.4rem;
  }

  .named-weapon {
    background: var(--namedGold);
  }

  .exotic-weapon {
    background: var(--exoticOrange);
  }

  @media (max-width: 35em) {
    padding: 0;
    .single-weapon__content {
      padding-top: 1rem;
      padding-bottom: 1rem;
      display: flex;
      flex-direction: column;
    }

    .single-weapon__sub-content {
      flex-direction: column;
    }
  }
`;

export const SINGLE_WEAPON_QUERY = gql`
  query SINGLE_WEAPON_QUERY($id: ID!) {
    allWeapons(where: { id: $id }) {
      id
      class
      model
      family
      magazineSize
      rpm
      modSlots
      reloadSpeed
      reloadSpeedFromEmpty
      accuracy
      stability
      optimalRange
      maxRange
      headshotMultiplier
      weaponBonusType
      maxBonusValue
      weaponBonusTypeTwo
      maxBonusTwoValue
      damageLevel40
      damageWt5
      flavourText
      isNamed
      isExotic
      weaponTalent {
        id
        name
        image {
          id
          image {
            publicUrlTransformed
          }
        }
      }
      exoticWeaponAttachment {
        id
        name
        attachmentSlot
        opticsSlotBonusType
        opticsBonusValueLevel40
        opticsValueWT5
        magazineSlotBonusType
        magazineValueLevel40
        magazineValueWT5
        underbarrelSlotBonusType
        underbarrelValueLevel40
        underbarrelValueWT5
        muzzleSlotBonusType
        muzzleValueLevel40
        muzzleValueWT5
      }
      averageWeapon {
        id
        class
        classBonusType
        maxClassBonusValue
        modSlots
        rpm
        magazineSize
        headshotMultiplier
        accuracy
        stability
        optimalRange
        maxRange
        reloadSpeed
        reloadSpeedFromEmpty
        damageLevel40
        damageWT5
        weaponTalent {
          id
          name
          image {
            id
            image {
              publicUrlTransformed
            }
          }
        }
        weaponThirdAttribute {
          id
          attribute
          valueLevel40
          valueWT5
          classARAttribute
          classLMGAttribute
          classMMRAttribute
          classPistolAttribute
          classRifleAttribute
          classShotgunAttribute
          classSMGAttribute
        }
      }
    }
  }
`;

export default function SingleWeapon({ weapon }) {
  return (
    <>
      <div className="single-weapon__content">
        <div className="single-weapon__details">
          <div />
          <table>
            <thead className="single-weapon__title-bar">
              <tr>
                <th className="single-weapon__subheading">Base Damage</th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <th scope="row">Level 40</th>
                <td>{humanReadableNumber(weapon.damageLevel40)}</td>
              </tr>
              <tr>
                <th scope="row">World Tier 5</th>
                <td>{humanReadableNumber(weapon.damageWt5) || 'N/A'}</td>
              </tr>
            </tbody>
            <thead className="single-weapon__title-bar">
              <tr>
                <th className="single-weapon__subheading">Specifications</th>
              </tr>
            </thead>

            <tbody>
              {(!weapon.accuracy || !weapon.stability) && (
                <>
                  <tfoot>
                    <tr>
                      <td colSpan="2">
                        <small>
                          0 indicates no community-sourced value available
                        </small>
                      </td>
                    </tr>
                  </tfoot>
                </>
              )}

              {weapon.maxBonusValue > 1 && (
                <>
                  <tr>
                    <th scope="row">{deAbbreviate(weapon.weaponBonusType)}</th>
                    <td>{weapon.maxBonusValue} %</td>
                  </tr>
                </>
              )}

              {weapon.maxBonusTwoValue > 1 && (
                <>
                  <tr>
                    <th scope="row" data-testid="secondDamageBonusTest">
                      {deAbbreviate(weapon.weaponBonusTypeTwo)}
                    </th>
                    <td>{displayPercentage(weapon.maxBonusTwoValue)} %</td>
                  </tr>
                </>
              )}

              <tr>
                <th scope="row">Headshot Multiplier</th>
                <td>x {hsDisplay(weapon.headshotMultiplier)}</td>
              </tr>
              <tr>
                <th scope="row">Rate of Fire</th>
                <td>{weapon.rpm} rounds per minute</td>
              </tr>
              <tr>
                <th scope="row">Magazine</th>
                <td>{weapon.magazineSize}</td>
              </tr>

              {weapon.modSlots === 0 && (
                <>
                  <tr>
                    <th scope="row">Mod Slots</th>
                    <td>Nil adjustable</td>
                  </tr>
                </>
              )}

              {weapon.modSlots > 0 && (
                <>
                  <tr>
                    <th scope="row">Mod Slots</th>
                    <td>{weapon.modSlots}</td>
                  </tr>
                </>
              )}

              <tr>
                <th scope="row">Accuracy</th>
                <td>{displayPercentage(weapon.accuracy)} %</td>
              </tr>
              <tr>
                <th scope="row">Stability</th>
                <td>{displayPercentage(weapon.stability)} %</td>
              </tr>
              <tr>
                <th scope="row">Optimal Range</th>
                <td>{weapon.optimalRange} m</td>
              </tr>
              <tr>
                <th scope="row">Max Range</th>
                <td>{weapon.maxRange} m</td>
              </tr>
              <tr>
                <th scope="row">Reload Speed</th>
                <td>{millisecondsToSeconds(weapon.reloadSpeed)} seconds</td>
              </tr>
              <tr>
                <th scope="row">Reload Speed (empty magazine)</th>
                <td>
                  {millisecondsToSeconds(weapon.reloadSpeedFromEmpty)} seconds
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </>
  );
}
