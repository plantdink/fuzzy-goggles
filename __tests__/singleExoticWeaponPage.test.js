import { render, screen, cleanup } from '@testing-library/react';
import { MockedProvider } from '@apollo/react-testing';
import { fakeExoticWeaponQuery } from '../lib/testUtils';
import { SINGLE_WEAPON_QUERY } from '../components/SingleWeapon';
import SingleExoticWeaponPage from '../pages/weapons/exoticWeapon/[id]';

afterEach(cleanup);

const fakeQuery = fakeExoticWeaponQuery();

const mocks = [
  {
    request: {
      query: SINGLE_WEAPON_QUERY,
      variables: {
        id: fakeQuery.id,
      },
    },
    result: {
      data: {
        allWeapons: [fakeQuery],
      },
    },
  },
];

const errorMocks = [
  {
    request: {
      query: SINGLE_WEAPON_QUERY,
      variables: {
        id: fakeQuery.id,
      },
    },
    result: {
      errors: [
        {
          message: 'Weapon not found.',
        },
      ],
    },
  },
];

describe('<SingleExoticWeaponPage />', () => {
  it('displays loading when the page first renders', async () => {
    const { container, debug } = render(
      <>
        <MockedProvider mocks={mocks}>
          <SingleExoticWeaponPage query={{ id: `${fakeQuery.id}` }} />
        </MockedProvider>
      </>
    );

    expect(screen.getByText('Loading...')).toBeInTheDocument();
  });

  it('loads the data and renders all components with data correctly for a single exotic weapon page', async () => {
    const { container, debug, findByText } = render(
      <>
        <MockedProvider mocks={mocks}>
          <SingleExoticWeaponPage query={{ id: `${fakeQuery.id}` }} />
        </MockedProvider>
      </>
    );

    await screen.findByTestId('singleWeaponTest');
    expect(container).toMatchSnapshot();
    // debug();

    const testTitle = await findByText(fakeQuery.model);
    expect(testTitle).toBeInTheDocument();

    const testValueWarning = screen.findByText(
      /note: 0 indicates no value available/i
    );
    if (
      (fakeQuery.accuracy ||
        fakeQuery.stability ||
        fakeQuery.optimalRange ||
        fakeQuery.maxRange) === 0
    )
      expect(testValueWarning).toBeInTheDocument();

    const exoticAttachmentComponent = await screen.findByTestId(
      'exoticWeaponTest'
    );
    if (fakeQuery.isExotic === 'YES')
      expect(exoticAttachmentComponent).toBeInTheDocument();

    const secondDamageBonus = await screen.findByTestId(
      'secondDamageBonusTest'
    );
    if (fakeQuery.weaponBonusTypeTwo !== null)
      expect(secondDamageBonus).toBeInTheDocument();

    const screenLinks = Array.from(container.querySelectorAll('a'));
    const talentLinks = fakeQuery.weaponTalent;
    expect(screenLinks.length).toEqual(talentLinks.length);
  });

  it('does not render elements for attributes that are null or empty', async () => {
    const { container, debug } = render(
      <>
        <MockedProvider mocks={mocks}>
          <SingleExoticWeaponPage query={{ id: `${fakeQuery.id}` }} />
        </MockedProvider>
      </>
    );
    await screen.findByTestId('singleWeaponTest');

    // dodgy way of checking if a value DOES NOT render
    // TODO: Find a better way to check this
    const secondDamageBonusValue = fakeQuery.maxBonusTwoValue;
    if (fakeQuery.weaponBonusTypeTwo === null)
      expect(secondDamageBonusValue).not.toBeInTheDocument();
  });

  it('Throws an error when an item is not found', async () => {
    const { container, debug } = render(
      <>
        <MockedProvider mocks={errorMocks}>
          <SingleExoticWeaponPage query={{ id: `${fakeQuery.id}` }} />
        </MockedProvider>
      </>
    );

    await screen.findByTestId('graphql-error');
    expect(container).toHaveTextContent('Fucks sake!!');
    expect(container).toHaveTextContent('Weapon not found.');
  });
});
