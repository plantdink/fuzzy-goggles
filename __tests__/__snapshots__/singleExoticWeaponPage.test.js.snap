// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleExoticWeaponPage /> loads the data and renders all components with data correctly for a single exotic weapon page 1`] = `
<div>
  <div
    class="SingleWeapon__SingleWeaponStyles-sc-1l6wyg4-0 iQbJbJ"
    data-testid="singleWeaponTest"
  >
    <div
      class="single-weapon__title-bar"
    >
      <h1
        class="single-weapon__heading"
      >
        Test Fake Exotic
      </h1>
    </div>
    <p>
      Assault Rifle
       
      AK-47
    </p>
    <div
      class="single-weapon__details"
    >
      <div
        class="single-weapon__title-bar"
      >
        <h3
          class="single-weapon__subheading"
        >
          Stats Comparison
        </h3>
      </div>
    </div>
    <div
      class="single-weapon__title-bar"
    >
      <h3
        class="single-weapon__section-heading"
      >
        Weapon Stats
      </h3>
    </div>
    <blockquote />
    <div
      class="single-weapon__sub-content"
    >
      <div
        class="single-weapon__content"
      >
        <div
          class="single-weapon__details"
        >
          <div />
          <table>
            <thead
              class="single-weapon__title-bar"
            >
              <tr>
                <th
                  class="single-weapon__subheading"
                >
                  Base Damage
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <th
                  scope="row"
                >
                  Level 40
                </th>
                <td>
                  60,645
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  World Tier 5
                </th>
                <td>
                  13,437
                </td>
              </tr>
            </tbody>
            <thead
              class="single-weapon__title-bar"
            >
              <tr>
                <th
                  class="single-weapon__subheading"
                >
                  Specifications
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <th
                  scope="row"
                >
                  Damage to Armour
                </th>
                <td>
                  21
                   %
                </td>
              </tr>
              <tr>
                <th
                  data-testid="secondDamageBonusTest"
                  scope="row"
                >
                  Damage to Health
                </th>
                <td>
                  9.5
                   %
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Headshot Multiplier
                </th>
                <td>
                  x 
                  1.55
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Rate of Fire
                </th>
                <td>
                  600
                   rounds per minute
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Magazine
                </th>
                <td>
                  30
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Mod Slots
                </th>
                <td>
                  4
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Accuracy
                </th>
                <td>
                  47.7
                   %
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Stability
                </th>
                <td>
                  64
                   %
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Optimal Range
                </th>
                <td>
                  0
                   m
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Max Range
                </th>
                <td>
                  68
                   m
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Reload Speed
                </th>
                <td>
                  2.20
                   seconds
                </td>
              </tr>
              <tr>
                <th
                  scope="row"
                >
                  Reload Speed (empty magazine)
                </th>
                <td>
                  2.70
                   seconds
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
      <div
        class="single-weapon__details"
        data-testid="exoticWeaponTest"
      >
        <div
          class="single-weapon__title-bar"
        >
          <h3
            class="single-weapon__subheading"
          >
            Weapon Attachments
          </h3>
        </div>
        <table>
          <thead>
            <tr>
              <th
                scope="col"
              >
                Slot
              </th>
              <th
                scope="col"
              >
                Name
              </th>
              <th
                scope="col"
              >
                Level 40
              </th>
              <th
                scope="col"
              >
                World Tier 5
              </th>
            </tr>
          </thead>
          <tfoot>
            <tr>
              <td
                colspan="4"
              >
                <small>
                  * indicates weapon not available at World Tier 5
                </small>
              </td>
            </tr>
          </tfoot>
          <tbody>
            <tr>
              <th
                scope="row"
              >
                Optics
              </th>
              <td>
                Fake Exotic Attachment 1
              </td>
              <td>
                100
              </td>
              <td>
                *
              </td>
            </tr>
            <tr>
              <th
                scope="row"
              >
                Magazine
              </th>
              <td>
                Fake Exotic Attachment 2
              </td>
              <td>
                100
              </td>
              <td>
                *
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
    <div
      class="single-weapon__details"
      data-testid="testARNotes"
    >
      <div
        class="single-weapon__title-bar"
      >
        <h2
          class="single-weapon__section-heading"
        >
          Class Description
        </h2>
      </div>
      <p>
        An assault rifle is a short, compact, selective-fire weapon that fires a cartridge (typically either 5.56mm or 7.62mm) that has more power than a pistol or submachine gun but less power than a standard rifle or battle rifle. An assault rifle is designed for medium to long range engagements however it will perform well in CQB depending on the agents skill.
      </p>
      <p>
        Assault rifles are either a standard configuration with the magazine located in front of the pistol grip, or a "bullpup" with the magazine located behind the pistol grip of the weapon. Assault rifles can also be fitted with a range of attachments including extended magazines, sights, fore-grips and muzzle attachments, like a suppressor.
      </p>
    </div>
    <div
      class="single-weapon__details"
    >
      <div
        class="single-weapon__title-bar"
      >
        <h1
          class="single-weapon__section-heading"
        >
          Compatible Weapon Talents
        </h1>
      </div>
      <div
        class="LinkSmallBrand__LinkStyles-sc-1sivfgj-0 ljFRYs"
      >
        <h3
          class="LinkSmallBrand__TitleLink-sc-1sivfgj-1 gKRdRa"
        >
          <a
            href="/talents/weaponTalent/lnulbvclbcgl"
          >
            Fake Weapon Talent
          </a>
        </h3>
        <img
          class="standard-item"
          src="FakeWeapon.png"
        />
      </div>
    </div>
  </div>
</div>
`;
